### PR TITLE
Add version constraint to numpy

### PR DESCRIPTION
**Pull request by y0z**
## Motivation
- Fix CI error caused by numpy 2.0.0.

## Description of the changes

- Add version constraint to numpy for CIs broken

### CODE REVIEW

File: optuna_integration/catboost/requirements.txt [hunk 1]
```diff
@@ -1 +1,2 @@
 catboost
+numpy<2.0.0
```
Comment (nabenabe0928) [inline optuna_integration/catboost/requirements.txt:2]: Could you add this commen?
```suggestion
numpy<2.0.0  # TODO(nabe): Remove this constraint once this issue is resolved. https://github.com/numpy/numpy/issues/26710
```

File: optuna_integration/cma/requirements.txt [hunk 2]
```diff
@@ -1 +1,2 @@
 cma
+numpy<2.0.0
```
Comment (nabenabe0928) [inline optuna_integration/cma/requirements.txt:2]: Same here

File: optuna_integration/shap/requirements.txt [hunk 3]
```diff
@@ -1 +1,2 @@
+numpy<2.0.0
```
Comment (nabenabe0928) [inline optuna_integration/shap/requirements.txt:1]: Same here

**Review by nabenabe0928 — COMMENTED** [review]
Thank you for the PR, I left minor comments!

**Comment by y0z** [comment]
@nabenabe0928 

I applied your comments. PTAL.

**Review by nabenabe0928 — APPROVED** [review]
LGTM!

**Comment by y0z** [comment]
Related: https://github.com/optuna/optuna/issues/5502

**Pull request merged**